compiler/hie-core/src/Development/IDE/LSP/LanguageServer.hs

```diff
@@ -89,18 +100,20 @@ runLanguageServer getIdeState = do
 
 -- | Things that get sent to us, but we don't deal with.
 --   Set them to avoid a warning in VS Code output.
-setHandlersIgnore :: WithMessage -> LSP.Handlers -> IO LSP.Handlers
-setHandlersIgnore _ x = return x
+setHandlersIgnore :: PartialHandlers
+setHandlersIgnore = PartialHandlers $ \_ x -> return x
     {LSP.cancelNotificationHandler = none
     ,LSP.initializedHandler = none
-    ,LSP.codeLensHandler = none -- FIXME: Stop saying we support it in 'options'
     }
     where none = Just $ const $ return ()
 
 
-mergeHandlers :: [WithMessage -> LSP.Handlers -> IO LSP.Handlers] -> WithMessage -> LSP.Handlers -> IO LSP.Handlers
-mergeHandlers = foldl f (\_ a -> return a)
-    where f x1 x2 r a = x1 r a >>= x2 r
+setHandlersExit :: PartialHandlers
+setHandlersExit = PartialHandlers $ \WithMessage{..} x -> return x
```

Comment:
The default handler here is quite reasonable so I’m not sure we need to override it https://github.com/alanz/haskell-lsp/blob/d00f83700cdc681f65015c7343206c134919d15f/src/Language/Haskell/LSP/Core.hs#L309

Reply:
Ah, didn't see the default, will delete. FWIW, before we captured the close notification and shut down on that, which is definitely the wrong thing to do...

Reply:
Yeah that was left over from before we switched to `haskell-lsp`. We never actually got to that code path afaik.

Reply:
My reading was we would - when the shutdown message came in we exited, and VS Code is meant to send shutdown first, then exit. Either way, dropping this code is much nicer.